server/src/main/scala/de/frosner/broccoli/nomad/models/Allocation.scala

```diff
@@ -2,20 +2,31 @@ package de.frosner.broccoli.nomad.models
 
 import play.api.libs.functional.syntax._
 import play.api.libs.json.{JsPath, Reads}
+import shapeless.tag
+import shapeless.tag.@@
 
 /**
   * A partial model for a single allocations.
   */
-final case class Allocation(id: String,
-                            nodeId: String,
-                            clientStatus: ClientStatus,
-                            taskStates: Map[String, TaskStateEvents])
+final case class Allocation(
+    id: String @@ Allocation.Id,
+    jobId: String @@ Job.Id,
```

Comment:
Personally I don't need them, I find they introduce a lot of clutter and boilerplate without any significant benefit.  These types are short; aliasing them won't save much will it?